src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -30,104 +28,119 @@ public UnityEventProcessor(IApplication? application = null)
+            sdk.AddPackage(UnitySdkInfo.PackageName, UnitySdkInfo.Version);
+            sdk.Name = UnitySdkInfo.Name;
+            sdk.Version = UnitySdkInfo.Version;
+        }
 
-            @event.Contexts.App.StartTime = DateTimeOffset.UtcNow.AddSeconds(-Time.realtimeSinceStartup);
+        private static void PopulateApp(App app)
+        {
+            app.StartTime = DateTimeOffset.UtcNow
+                // NOTE: Time API requires main thread
+                .AddSeconds(-Time.realtimeSinceStartup);
 
-            // TODO:  @event.Contexts["Unity"] = new UnityContext (values to be read on the main thread)
-            @event.SetExtra("unity:processorCount", SystemInfo.processorCount.ToString());
-            @event.SetExtra("unity:supportsVibration", SystemInfo.supportsVibration.ToString());
-            @event.SetExtra("unity:installMode", Application.installMode.ToString());
+            app.BuildType = Debug.isDebugBuild ? "debug" : "release";
+        }
 
+        private static void PopulateOperatingSystem(OperatingSystem operatingSystem)
+        {
             // TODO: Will move to raw_description once parsing is done in Sentry
-            @event.Contexts.OperatingSystem.Name = SystemInfo.operatingSystem;
+            operatingSystem.Name = SystemInfo.operatingSystem;
+        }
 
-            switch (Input.deviceOrientation)
-            {
-                case UnityEngine.DeviceOrientation.Portrait:
-                case UnityEngine.DeviceOrientation.PortraitUpsideDown:
-                    @event.Contexts.Device.Orientation = DeviceOrientation.Portrait;
-                    break;
-                case UnityEngine.DeviceOrientation.LandscapeLeft:
-                case UnityEngine.DeviceOrientation.LandscapeRight:
-                    @event.Contexts.Device.Orientation = DeviceOrientation.Landscape;
-                    break;
-                case UnityEngine.DeviceOrientation.FaceUp:
-                case UnityEngine.DeviceOrientation.FaceDown:
-                    // TODO: Add to protocol?
-                    break;
-            }
+        private static void PopulateDevice(Device device, IApplication application)
+        {
+            device.ProcessorCount = SystemInfo.processorCount;
+            device.SupportsVibration = SystemInfo.supportsVibration;
+            device.BatteryStatus = SystemInfo.batteryStatus.ToString();
+            device.DeviceType = SystemInfo.deviceType.ToString();
+            device.CpuDescription = SystemInfo.processorType;
+            device.Timezone = TimeZoneInfo.Local;
+            device.ProcessorCount = SystemInfo.processorCount;
+            device.SupportsVibration = SystemInfo.supportsVibration;
+            device.Name = SystemInfo.deviceName;
+
+            // This is the approximate amount of system memory in megabytes.
+            // This function is not supported on Windows Store Apps and will always return 0.
+            device.MemorySize = SystemInfo.systemMemorySize;
```

Comment:
Good comment. What about we check if it's `0` and avoid adding it altogether?

Reply:
I noticed that it's used 2 times and we already have the logic to check against `0`, will remove that one altogether)